cmake/compiler/gcc.cmake

```diff
@@ -138,5 +138,13 @@ endif()
 foreach(isystem_include_dir ${NOSTDINC})
   list(APPEND isystem_include_flags -isystem "\"${isystem_include_dir}\"")
 endforeach()
-set(CMAKE_REQUIRED_FLAGS -nostartfiles -nostdlib ${isystem_include_flags} -Wl,--unresolved-symbols=ignore-in-object-files)
+# The CMAKE_REQUIRED_FLAGS variable is used by check_c_compiler_flag()
+# (and other commands which end up calling check_c_source_compiles())
+# to add additional compiler flags used during checking. These flags
+# are unused during "real" builds of Zephyr source files linked into
+# the final executable.
+#
+# Appending onto any existing values lets users specify
+# toolchain-specific flags at generation time.
+set(CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS} -nostartfiles -nostdlib ${isystem_include_flags} -Wl,--unresolved-symbols=ignore-in-object-files)
```

Comment:
list(APPEND

A part from that, this patch looks sane.

Reply:
Much better.